javascript/javascript1/week4/homework/voiceAssistant.js

```diff
@@ -0,0 +1,155 @@
+// Global variables to store the user's name and todos.
+let userName = "";
+let todos = [];
+
+function getReply(command) {
+  const cmd = command.trim();
+  const cmdLower = cmd.toLowerCase();
+
+  // 1. "Hello my name is Benjamin"
+
+  const greetingPhrase = "hello my name is ";
+  if (cmdLower.startsWith(greetingPhrase)) {
+    const name = cmd.substring(greetingPhrase.length).trim();
+
+    if (userName && userName.toLowerCase() === name.toLowerCase()) {
+      return `You've already told me your name is ${userName}`;
+    } else {
+      userName = name;
+      return `Nice to meet you ${userName}`;
+    }
+  }
+
+  // 2. "What is my name?"
+  if (cmdLower.startsWith("what is my name")) {
+    if (userName) {
+      return `Your name is ${userName}`;
+    } else {
+      return "I don't know your name yet.";
+    }
+  }
+
+  // 3. "Add fishing to my todo" (or similar)
+
+  if (cmdLower.startsWith("add ") && cmdLower.indexOf(" to my todo") !== -1) {
+    const startIndex = "add ".length;
```

Comment:
Suggestion: strings are pretty easy to get wrong. Save into variables while possible.